enhancements/sig-architecture/58-addon-placement-annotation/README.md

```diff
@@ -0,0 +1,295 @@
+	// +optional
+	AddOnConfiguration ConfigCoordinates `json:"addOnConfiguration"`
+}
+
+// ConfigCoordinates represents the information for locating the CRD and CR that configures the add-on.
+type ConfigCoordinates struct {
+	// crdName is the name of the CRD used to configure instances of the managed add-on.
+	// This field should be configured if the add-on have a CRD that controls the configuration of the add-on.
+	// +optional
+	CRDName string `json:"crdName"`
+
+	// crName is the name of the CR used to configure instances of the managed add-on.
+	// This field should be configured if add-on CR have a consistent name across the all of the ManagedCluster instaces.
+	// +optional
+	CRName string `json:"crName"`
+
+	// lastObservedGeneration is the observed generation of the custom resource for the configuration of the addon.
+	// +optional
+	LastObservedGeneration int64 `json:"lastObservedGeneration,omitempty"`
+}
+```
+
+but the `ConfigCoordinates` has some limits, it only support to use `CustomResourceDefinition` as the addon
+configuration, and for addon developers, it's not easy to locate the configuration object in their controllers. we want
+to refactor the `ConfigCoordinates` to
+
+``` golang
+// ConfigCoordinates contains enough information to let one addon identify its configuration object.
+// A config object must be in the same namespace with its addon, or be cluster-scoped, so there is no
+// namespace field.
+type ConfigCoordinates struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion"`
+	// Kind of the referent.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `json:"kind"`
+	// Name of the referent.
+	// More info: http://kubernetes.io/docs/user-guide/identifiers#names
+	Name string `json:"name"`
+}
+```
+
+with this change, the `ConfigCoordinates` will be able to support all types of configuration, and the developers will
+be able to get the `GroupVersionResource` of the configuration object.
+
+In the `ManagedClusterAddOn` API, we will also introduce `AddOnConfiguration` option, When addon has its own
+configuration, the addon can use this option to locate its configuration
+
+```golang
+// ManagedClusterAddOnSpec defines the install configuration of
+// an addon agent on managed cluster.
+type ManagedClusterAddOnSpec struct {
+	// installNamespace is the namespace on the managed cluster to install the addon agent.
+	// If it is not set, open-cluster-management-agent-addon namespace is used to install the addon agent.
+	// +optional
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	InstallNamespace string `json:"installNamespace,omitempty"`
+
+	// addOnConfiguration is a configuration object reference that is used by current addon.
+	// +optional
+	AddOnConfiguration ConfigCoordinates `json:"addOnConfiguration,omitempty"`
```

Comment:
we should only have namespace/name here. Also we need update status field

Reply:
yes, we need update status